### PR TITLE
Make the Python3 branch completely flake8 and isort complicance

### PR DESCRIPTION
**Pull request by loechel**
This branch also adds config on travis-ci and tox to always run those tests

### CODE REVIEW

**Comment by hannosch** [comment]
-1 from me.

I think the common standard for imports in Zope (and Python in general) is to sort imports into three groups. First a group for imports from the standard library, a second group for imports from external dependencies and a third group for imports from the same project. This helps to make it clear where functionality is coming from and also helps in avoiding circular imports, as dependencies are imported before everything else. I also don't see the point of having a single import per line, and duplicating the prefix over and over again. Especially if those are backwards compatibility imports, so the chance of changing a single one of them and producing diff noise is low.

Taking AuthEncoding as an example, I don't mind the current state:
```
from AuthEncoding.AuthEncoding import (  # NOQA
    constant_time_compare, PasswordEncryptionScheme, registerScheme,
    listSchemes, SSHADigestScheme, SHADigestScheme, MySQLDigestScheme,
    pw_validate, is_encrypted, pw_encrypt, pw_encode)
```
If this would be non backwards-compatibility code, I'd write it as:
```
from AuthEncoding.AuthEncoding import (  # NOQA
    constant_time_compare,
    is_encrypted,
    listSchemes,
    MySQLDigestScheme,
    PasswordEncryptionScheme,
    pw_encode,
    pw_encrypt,
    pw_validate,
    registerScheme,
    SHADigestScheme,
    SSHADigestScheme,
)
```
This ensures adding / removing one of those lines results in a small diff file, without repeating the same prefix.

This PR also comments out some unused imports. But these might be used by other packages or other places in the code. I'd rather keep them with a NOQA marker as before or put in a proper zope.deferredimport deprecation.

On a more general note, I think isort is dangerous, as there are too many places in Zope where import order matters and circular import problems are common.

**Comment by loechel** [comment]
Well, I think it is clear that I disagree to @hannosch at that point, as I have done this Pull Request.

I think we should devide two topics:

* isort --> Tool and Sorting of Imports
* flake8 --> Checking and enforcing Coding Conventions

As we did start this discussion face to face, this should be also a kind of documentation.

Coding conventions are something we both welcome. Code, especially Python Code should be easyly readable to help understand it. As Zope Packages are older than PEP8 and PEP20 most of the naming conventions did not match, but as PEP8 states that if Project specific Code Conventions exists those have precedence.

This Pull Request is about enforcing Code Conventions. Therefore the tox and travis-ci configuration have corresponding sections for flake8 and isort (actually 2, isort --> isort in check only mode, isort-apply --> allways commented out in active parts, it could be used to apply the sort and grouping settings)

To the Code Convention for me belongs also rules about imports. 
We did start by applying Plone Code Conventions (https://docs.plone.org/develop/styleguide/python.html) as those are the only documented which in Zope/Plone Context exists today.

One specific part of it is Grouping and sorting Imports (https://docs.plone.org/develop/styleguide/python.html#about-imports)
For me it doesn't matter which rules for grouping and sorting are applied, but it should be a defined rule. 
For RestrictedPython @icemac and me agreed on applying the Plone standard rules.

The isort rules (the are just checked by the tests, not automatically applied) are defined in setup.cfg [isort], documentation about how to configure could be found at http://isort.readthedocs.io/en/latest/#configuring-isort 
 
The other point @hannosch had, is that the import order in Zope Context could sometimes be very problamatic as it produced circular dependencies. Also Moved Implementations and backwards compatibility imports might be warned about. Both is no problem in this context, as enforcing the code conventions allows you to mark specific test(e.g. # NOQA: E401  # BBB import) so that no warning is shown. This also enforces developers to think and comment why this is in a specific order or imported at all. 

In summary I am ok to discuss which isort rules should be applied, but I strongly request to enforce code conventions.

**Comment by loechel** [comment]
Also about code conventions, we do add .editorconfig config files to all packages, so that configured editors also apply those conventions (indent, empty line at bottom of file, no whitespaces in empty lines) automatically and flake8 won't warn about silly unnecessary differences.

**Comment by loechel** [comment]
after an offline discussion ( @hannosch @icemac @loechel ) did agreed that we want flake8 and isort in check mode on AccessControll and other Zope Packages (RestrictedPython, ...) 

for Zope Packages we go with the pep8 import rules (https://www.python.org/dev/peps/pep-0008/#imports) to seperate stdlib, thirdparty and own imports. 

I will adjust the setup.cfg [isort] section and apply the changes on both packages.

**Comment by loechel** [comment]
@hannosch @icemac now ready for review.

**Comment by hannosch** [comment]
I'm with icemac on this one.

The flake8 changes alone would probably be fine in a separate PR, though for Zope code a global E301 exception makes more sense to me. We have these security related pseudo decorators all over the place and adding a comment to each of them adds a lot of line noise.

isort in general seems to add little value with a lot of bookkeeping. I honestly prefer to stick to the rough PEP8 guideline of separating imports into three groups, as that is what most of the code base does. Otherwise the general principle of adjusting to what you find in the file seems to be a good enough approach. The Zope Toolkit code style guidelines are also fairly silent on this point (http://zopetoolkit.readthedocs.io/en/latest/codingstyle/index.html). Though those never really applied to Zope 2 related code, they seem to make sense to me.

When we adopted flake8/pep8 it added a lot of whitespace/reformatting commits. Those were already painful, but I think they actually help readability and ongoing code maintenance. I can't say the same about isort. It adds another large set of commits/changes without really aiding code readability. And as we've seen with some of those attempts, there are imports in Zope files for compatibility purposes all over the place, and removing "unused" imports is fairly tricky.

At this point, I'd rather we focus our effort and time on porting to Python 3 and not spend our time arguing about import sorting.
